db_master_deploy/deploy.sh

```diff
@@ -370,6 +371,52 @@ copy_table() {
     fi
 }
 
+#######################################
+# write lock
+# Globals:
+#   target_db
+#   COMMAND
+#   USER
+# Arguments:
+#   None
+# Returns:
+#   None
+#######################################
+write_lock() {
+    local lockfile="${LOCK_DIR}/${target_db}.lock"
+    # if target db is locked enter loop and wait for lockfile
+    until [ ! -f ${lockfile} ]
+    do
+        echo "target db ${target_db} is locked, waiting for deploy process to finish"
+        sleep 5
+    done
```

Comment:
Shouldn't we have something like max retries?

Otherwise, this might congest up if something goes wrong with the lock file.